operator/pkg/model/translation/gateway-api/translator.go

```diff
@@ -110,7 +111,15 @@ func (t *gatewayAPITranslator) Translate(m *model.Model) (*ciliumv2.CiliumEnvoyC
 		lbSvc.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicy("")
 	}
 
-	return cec, lbSvc, getEndpoints(*source), err
+	shortedName := model.Shorten(source.Name)
```

Comment:
All these "decorating" aspects (shorten name, adding labels and annotations) to the CEC should be moved into it's own function.

Even if moved to into it's own method, i would avoid having these nested calls to `mergeMap`

Reply:
good point